README: add repo logo, downloads badge, center badges

dimitryzub opened the pull request:
Changes in this PR look like this:

![image](https://user-images.githubusercontent.com/78694043/233933030-fd83fd0e-66ef-405e-b0d3-ff1c4ff23c57.png)

Additional variations of the logo:

![serpapi-rust-repo-logo-02](https://user-images.githubusercontent.com/78694043/233932939-c809c007-dfae-41a7-8f78-472ed33ed5c6.svg)

![serpapi-rust-repo-logo-01](https://user-images.githubusercontent.com/78694043/233932933-9bb18ca7-7368-4425-a8ff-8cefb2b1201a.svg)

dimitryzub committed:
README: add repo logo, downloads badge, center badges

dimitryzub committed:
README: fix downloads badge link

jvmvik commented:
I prefer the last variation. can you update the readme ?
<img width="412" alt="image" src="https://user-images.githubusercontent.com/330204/235414715-00aec0e9-dc80-4853-875e-a238022d7f51.png">

dimitryzub committed:
README: update repository logo

This change takes third variation of the logo:
- https://github.com/serpapi/serpapi-search-rust/pull/1#issue-1680658937

dimitryzub commented:
@jvmvik should be replaced now 👍

I've also replaced SVG with PNG as SVG has aliasing artifacts on the lines due to how Adobe Illustrator traces PNG images to SVG. PNG image don't have such artifacts.
